### PR TITLE
rpc: update some RPCExamples to bech32

### LINKED ISSUE #18185: Change address format in RPCExamples from legacy to bech32
Regarding that Bitcoin Core switched to bech32 as the default format for addresses recently (with release 0.19.0.1), this should also be reflected in the RPC help texts. The change was already done for the RPCs `getaddressinfo` (commit https://github.com/bitcoin/bitcoin/commit/2ee0cb3330ccf70f0540cb42370796e32eff1569 by jonatack, PR https://github.com/bitcoin/bitcoin/pull/17283) and `validateaddress` (commit https://github.com/bitcoin/bitcoin/pull/18122/commits/7f1475c7119e8c72bce39a63386a6ca859066b80 by my humble self, PR https://github.com/bitcoin/bitcoin/pull/18122/), but there are still some RPCExamples around with Base58 addresses, i.e. 
- `setlabel`
- `sendtoaddress`
- `getreceivedbyaddress`
- .....

All those legacy address usages should be identified and substituted by bech32 addresses where it makes sense (note that e.g. for `signmessage` and `verifymessage` there is as far as I know no possibility to sign with segwit addresses yet, hence the legacy addresses should stay). For the address, the newly introduced string constant `EXAMPLE_ADDRESS` (in `src/rpc/util.h`) can be used, which contains an illegal bech32 address.

#### Useful skills:

Basic understanding of address formats.

#### Want to work on this issue?

The purpose of the `good first issue` label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.

You do not need to request permission to start working on this. You are encouraged to comment on the issue if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by yusufsahinhamza**
Update some RPC examples from legacy to bech32. 

Fixes #18185.

### CODE REVIEW

**Comment by fanquake** [comment]
cc @theStack

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1223,7 +1223,7 @@ static UniValue listreceivedbyaddress(const JSONRPCRequest& request)
                     HelpExampleCli("listreceivedbyaddress", "")
             + HelpExampleCli("listreceivedbyaddress", "6 true")
             + HelpExampleRpc("listreceivedbyaddress", "6, true, true")
-            + HelpExampleRpc("listreceivedbyaddress", "6, true, true, \"1M72Sfpbz1BPpXFHz9m3CdqATR44Jvaydd\"")
+            + HelpExampleRpc("listreceivedbyaddress", "6, true, true, " + EXAMPLE_ADDRESS + "")
```
Comment (Sjors) [inline src/wallet/rpcwallet.cpp]: The extra `+ ""` isn't necessary.

**Review by Sjors — APPROVED** [review]
tACK af23472cb39829f88868577f41510d0dc25f246a 

I would prefer to get #17809 merged first though.

File: src/wallet/rpcwallet.cpp (hunk 1 above)
Comment (Sjors) [reply src/wallet/rpcwallet.cpp]: Thanks for dropping that. Can you squash these two commits?

Comment (yusufsahinhamza) [reply src/wallet/rpcwallet.cpp]: @Sjors Done!

**Comment by Sjors** [comment]
re-ACK de83a30e7b7244e7cd7f9df7ad829aa96e479f3f

**Comment by theStack** [comment]
Code-Review ACK https://github.com/bitcoin/bitcoin/commit/de83a30e7b7244e7cd7f9df7ad829aa96e479f3f, changes LGTM

Note though that this PR alone does not completely fix #18185 yet (which is fine as the PR title only says "_some_ RPCExamples" and not _all_), as there are still some other RPCExamples containing legacy addresses:
- `sendmany`
- `addmultisigaddress`
- `listunspent`

Since those examples contain more than one address, it will probably make sense to add another constant for a second bech32 (invalid) address. I think it's okay to do all that in an extra follow-up PR.

**Comment by yusufsahinhamza** [comment]
@theStack Yes, actually i wasn't sure about adding another constant for a different address and EXAMPLE_ADDRESS isn't looks suitable for some examples because there is extra \ (back slash) before the " (quote) at the end of these addresses, that's why i left some of these examples:

addresses

And i think we can put these different example addresses to in a const array and use them like EXAMPLE_ADDRESS[0] and EXAMPLE_ADDRESS[1].

**Comment by theStack** [comment]
@yusufsahinhamza: Oh, thanks for pointing the backslash problem out -- considering that, it was not very wise from my side to include the quote characters in the `EXAMPLE_ADDRESS` string in https://github.com/bitcoin/bitcoin/commit/7f1475c7119e8c72bce39a63386a6ca859066b80 :thinking: I'd suggest then to remove the quotes from the constant and set the surrounding characters directly in the RPCExamples (not here, but in the course of a follow-up PR).

> And i think we can put these different example addresses to in a const array and use them like EXAMPLE_ADDRESS[0] and EXAMPLE_ADDRESS[1].

Agreed.

**Comment by theStack** [comment]
@yusufsahinhamza: It was not my intention that this PR gets closed, I think it would have been fine to merge it, since it already had two ACKs. Just because it doesn't fully fixes #18208 doesn't mean it's bad. With "follow-up PR" I meant a PR that is based on one that was already merged :-)

**Comment by yusufsahinhamza** [comment]
@theStack Oh okay, i thought we will continue in the new PR. Then it's better to merge first that one so we can really "follow" it :)

**Comment by MarcoFalke** [comment]
Given that you have already written the new pull and that all the lines touched here will be touched again in the other pull, it seems easier and less work for reviewers to just focus on that one. I'd suggest closing this one again.

**Pull request closed without merging**